src/optimizer/optimizer.cpp

```diff
@@ -182,6 +183,12 @@ unique_ptr<LogicalOperator> Optimizer::Optimize(unique_ptr<LogicalOperator> plan
 		compressed_materialization.Compress(plan);
 	});
 
+	// pushes LIMIT below PROJECTION
+	RunOptimizer(OptimizerType::LIMIT_PUSHDOWN, [&]() {
```

Comment:
I would put this before compressed materialization. The compressed materialization optimizer introduces extra projection operators. If the limit pushdown is above this, then the optimizer will avoid some extra push downs

Reply:
done!